File: engine/engine-model/src/main/scala/tools/packet/field/obstacle/EnvironmentMove.scala
package tools.packet.field.obstacle

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class EnvironmentMove(private var _environment: String, private var _mode: Int) extends PacketInput {
  def environment: String = _environment

  def mode: Int = _mode

  override def opcode(): SendOpcode = SendOpcode.FIELD_OBSTACLE_ONOFF
}

File: engine/engine-model/src/main/java/tools/packet/field/obstacle/EnvironmentMove.java
package tools.packet.field.obstacle;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record EnvironmentMove(String environment, Integer mode) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.FIELD_OBSTACLE_ON_OFF;
   }
}